gpflow/multioutput/conditionals.py

```diff
@@ -0,0 +1,451 @@
+    :param f: data matrix, M x P
+    :param full_cov: return the covariance between the datapoints
+    :param full_cov_output: return the covariance between the outputs.
+        Note: as we are using a independent kernel these covariances will be zero.
+    :param q_sqrt: matrix of standard-deviations or Cholesky matrices,
+        size M x P or P x M x M.
+    :param white: boolean of whether to use the whitened representation
+    :return: 
+        - mean:     N x P
+        - variance: N x P, P x N x N, N x P x P or N x P x N x P
+        Please see `gpflow.conditional.expand_independent_outputs` for more information
+        about the shape of the variance, depending on `full_cov` and `full_cov_output`.
+        
+    """
+    print("Conditional: SharedIndependentMof - SharedIndepedentMok")
+
+    Kmm = Kuu(feat, kern, jitter=settings.numerics.jitter_level)  # M x M
+    Kmn = Kuf(feat, kern, Xnew)  # M x N
+    if full_cov:
+        Knn = kern.K(Xnew, full_cov_output=False)[0, ...]  # N x N
+    else:
+        Knn = kern.Kdiag(Xnew, full_cov_output=False)[..., 0]  # N
+
+    fmean, fvar = base_conditional(Kmn, Kmm, Knn, f, full_cov=full_cov, q_sqrt=q_sqrt, white=white)  # N x P,  P x N x N or N x P
+    return fmean, expand_independent_outputs(fvar, full_cov, full_cov_output)
+
+
+@conditional.register(object, SeparateIndependentMof, SeparateIndependentMok, object)
+@conditional.register(object, SharedIndependentMof, SeparateIndependentMok, object)
+@conditional.register(object, SeparateIndependentMof, SharedIndependentMok, object)
+@name_scope("conditional")
+def _conditional(Xnew, feat, kern, f, *, full_cov=False, full_cov_output=False, q_sqrt=None, white=False):
+    """
+    Multi-output GP with independent GP priors.
+    Number of latent processes equals the number of outputs (L = P).
+
+    The covariance matrices used to calculate the conditional have the following shape:
+    - Kuu: P x M x M 
+    - Kuf: P x M x N
+    - Kff: P x N or P x N x N
+    
+    Further reference
+    -----------------
+    - See `gpflow.conditionals._conditional` for a detailed explanation of 
+      conditional in the single-output case.
+    - See the multiouput notebook for more information about the multiouput framework.
+    - See above for the parameters and the return value.
+    """
+
+    print("conditional: object, SharedIndependentMof, SeparateIndependentMok, object")
+    # Following are: P x M x M  -  P x M x N  -  P x N(x N)
+    Kmms = Kuu(feat, kern, jitter=settings.numerics.jitter_level)  # P x M x M
+    Kmns = Kuf(feat, kern, Xnew)  # P x M x N
+    kern_list = kern.kern_list if isinstance(kern, Combination) else [kern.kern] * len(feat.feat_list)
+    Knns = tf.stack([k.K(Xnew) if full_cov else k.Kdiag(Xnew) for k in kern_list], axis=0)
+    fs = tf.transpose(f)[:, :, None]  # P x M x 1
+    # P x 1 x M x M  or  P x M x 1
+    q_sqrts = tf.transpose(q_sqrt)[:, :, None] if q_sqrt.shape.ndims == 2 else q_sqrt[:, None, :, :]
```

Comment:
This part does not work with q_sqrt=None

Reply:
Correct, I'll fix that. Thanks